projects/novo-elements/src/elements/form/controls/BaseControl.ts

```diff
@@ -121,7 +121,7 @@ export class BaseControl extends ControlConfig {
     this.metaType = config.metaType;
     this.placeholder = config.placeholder || '';
     this.config = config.config || null;
-    this.dirty = !!config.value;
+    this.dirty = !!config.value || (config.dataType === 'Integer' && config.value !== null);
     this.multiple = !!config.multiple;
```

Comment:
what is the need to add a dataType specific check here?

Reply:
When the field is an integer and the default value is 0, it's not being marked dirty because the dirty flag evaluates to false. Then when the form is saved, that default int value doesn't get saved.

Reply:
i think we need to change the check for the `this.dirty` to make sure it doesn't incorrectly assume that `0` is an empty value